pkg/util/server/error.go

```diff
@@ -60,10 +63,22 @@ func ClientHTTPStatusAndError(err error) (int, error) {
 		return http.StatusBadRequest, err
 	case errors.Is(err, user.ErrNoOrgID):
 		return http.StatusBadRequest, err
+	case isRPC && (int(s.Code())/100 == 4 || int(s.Code())/100 == 5):
+		return int(s.Code()), errors.New(s.Message())
```

Comment:
This fix is a little weird as we are not testing with `error.Is`. Furthermore this method is used to determine whether a call should be retried. I think error handling needs some refacotring. The retry should not be based on the status code but with a method checking `error.Is`. Which leads us to deserializing errors. I do not know how that's possible.

Reply:
I also thing that the error handling needs some refactoring :)
But I also think that you cannot use `errors.Is` with `Status` errors, at least not to check whether they correspond to some of the types mentioned in this code.